src/encoder/mod.rs

```diff
@@ -238,13 +238,18 @@ impl<W: Write + Seek> TiffEncoder<W> {
 
     /// Convenience function to write an entire image from memory.
     pub fn write_image<C: ColorType>(&mut self, width: u32, height: u32, data: &[C::Inner]) -> TiffResult<()> where [C::Inner]: TiffValue {
+        if data.len() < (width * height) as usize {
```

Comment:
`width * height` could overflow `usize` on 32-bit systems. Please use [`checked_mul()`](https://doc.rust-lang.org/std/primitive.u32.html#method.checked_mul) instead.

Reply:
And convert `width` and `height` to `usize` before the multiplication.

Reply:
If the multiplication overflows (*without* first casting to `usize`), presumably an error should be returned because the image is too big for TIFF anyway (32 bit offsets)?

Reply:
Not necessarily. Compression, less then 8 bits per color, ... Just keep it separate, the check is cheap either way.

Reply:
Fair enough, I pushed an amend so the check is done now after a `usize` cast.